registry/remote/repository.go

```diff
@@ -114,6 +128,28 @@ func NewRepository(reference string) (*Repository, error) {
 	}, nil
 }
 
+// SetReferrersCapability sets the capability of Referrers API for the
+// repository. true: capable; false: not capable.
+// SetReferrersCapability is valid only when it is called for the first time.
+// - When the capability is set to true, the Referrers() function will always
+// request the Referrers API. Reference: https://github.com/opencontainers/distribution-spec/blob/v1.1.0-rc1/spec.md#listing-referrers
+// - When the capability is set to false, the Referrers() function will always
+// request the Referrers Tag. Reference: https://github.com/opencontainers/distribution-spec/blob/v1.1.0-rc1/spec.md#referrers-tag-schema
+// - When the capability is not set, the Referrers() function will automatically
+// determine which API to use.
+func (r *Repository) SetReferrersCapability(capable bool) {
+	if capable {
+		atomic.CompareAndSwapInt32(&r.referrersState, referrersStateUnknown, referrersStateSupported)
```

Comment:
The return value of `CompareAndSwapInt32()` is ignored but it should not. We may consider returning an error instead.

Reply:
Added an error.